controllers/chat/getSignedChannelList.js

```diff
@@ -40,7 +62,7 @@ const getSignedChannelList = async (req, res) => {
   const queriedChannels = [];
   const promisedChannels = [];
 
-  const client = GetstreamSingleton.getChatInstance();
+  const client = new StreamChat(process.env.API_KEY, process.env.API_SECRET);
   try {
     await client.connectUser({id: req.userId}, req.token);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

As mentioned earlier, creating a new instance of `StreamChat` for each request could lead to performance issues. Consider using a singleton pattern.

```diff
- const client = new StreamChat(process.env.API_KEY, process.env.API_SECRET);
+ const client = GetstreamSingleton.getChatInstance();
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->